core/src/test/scala/unit/kafka/zk/KafkaZkClientTest.scala

```diff
@@ -0,0 +1,97 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+package kafka.zk
+
+import kafka.zookeeper.ZooKeeperClient
+
+import org.junit.{After, Before, Test}
+import org.junit.Assert.{assertEquals, assertFalse, assertTrue}
+import org.apache.kafka.common.TopicPartition
+
+class KafkaZkClientTest extends ZooKeeperTestHarness {
+
+  private var zooKeeperClient: ZooKeeperClient = null
+  private var zkClient: KafkaZkClient = null
+
+  private val group = "my-group"
+  private val topicPartition = new TopicPartition("topic", 0)
+
+  @Before
+  override def setUp() {
+    super.setUp()
+    zooKeeperClient = new ZooKeeperClient(zkConnect, zkSessionTimeout, zkConnectionTimeout, null)
+    zkClient = new KafkaZkClient(zooKeeperClient, false)
+  }
+
+  @After
+  override def tearDown() {
+    zkClient.close()
+    zooKeeperClient.close()
```

Comment:
This seems unnecessary since zooKeeperClient.close() already calls zooKeeperClient.close().

Reply:
also we can wrap with CoreUtils.swallow() method.

Reply:
No, we should not do that in tests.

Reply:
Ok, I observed that is ZooKeeperTestHarness.  Should we remove  the usage of CoreUtils.swallow() from ZooKeeperTestHarness?